pkg/skaffold/util/util.go

```diff
@@ -154,6 +154,76 @@ func ReadConfiguration(filename string) ([]byte, error) {
 	}
 }
 
+// A type holding configuration properties
+type ConfigProperties map[string]string
+
+// Tests if a file exists
+func FileExists(filapath string) bool {
+	if _, err := os.Stat(filapath); !os.IsNotExist(err) {
+		if err == nil {
+			return true
+		}
+	}
+	return false
+}
+
+// Reads an environment file  that contains a set of environment variables
+func ReadEnvironmentConfigProperties(pathToSkaffoldFile string) (*viper.Viper, error) {
+	directory := filepath.Dir(pathToSkaffoldFile)
+	baseName := filepath.Base(pathToSkaffoldFile)
+	targetDirectory := ""
+	re := regexp.MustCompile("skaffold.*ya?ml")
+	if re.FindString(baseName) == "" {
+		targetDirectory = pathToSkaffoldFile
+	} else {
+		targetDirectory = directory
+	}
+	//
+	logrus.Debugf("Will load environment file from :%s", targetDirectory)
+	ret := viper.New()
+	ret.SetConfigType("yaml")
+	ret.SetConfigName("env")
+	ret.AddConfigPath("/etc/skaffold/")
+	ret.AddConfigPath("$HOME/.skaffold")
+	ret.AddConfigPath(targetDirectory)
+	err := ret.ReadInConfig()
+	if err != nil {
+		return ret, nil
+	}
+	return ret, nil
+}
+
+// Replace all the environment variables present in a skaffold configuration
+func ReplaceEnvironmentVariables(configurationSource *viper.Viper, skaffoldConfiguration []byte) []byte {
+	var patternMatcher = regexp.MustCompile(`\$[{]?[a-zA-Z.-]+[}]?`)
+	replacedBuffer := patternMatcher.ReplaceAllFunc(skaffoldConfiguration, func(bs []byte) []byte {
+		replacer := strings.NewReplacer("$", "", "{", "", "}", "")
+		currentMatch := replacer.Replace(string(bs))
+		configurationValue := configurationSource.GetString(currentMatch)
+		logrus.Debugf("Configuration file replace of %s by %s", currentMatch, configurationValue)
+		return []byte(configurationValue)
+	})
+	logrus.Debugf("Replaced configuration file:%s", string(replacedBuffer))
+	return replacedBuffer
+}
+
+// Inject  environment variables into skaffold configuration
+func InjectEnvironnmentVariables(pathToSkaffoldFile string, skaffoldConfiguration []byte, profiles []string) ([]byte, error) {
+	configurationSource, err := ReadEnvironmentConfigProperties(pathToSkaffoldFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "read environment config")
+	}
+	configurationSource.SetDefault("profile", strings.Join(profiles, ","))
```

Comment:
why do we need to set a value for `profile` here? shouldn't this get parsed normally through the replacer?

also, should it not be `profiles`?

Reply:
yes makes sense, will remove it